KML format util throws AssertionError

Occurs when the measurement tool is activated for an existing drawing layer which contains also a point feature.

```
AssertionError: `width` or `height` cannot be provided together with `scale`
    AssertionError AssertionError.js:78
    assert asserts.js:12
    Icon Icon.js:169
    clone Icon.js:259
    clone Style.js:236
    sanitizeStyle kml.js:48
    create kml.js:83
    create kml.js:72
    _save OlMeasurementHandler.js:620
    setSelectedAndSave
```

![grafik](https://user-images.githubusercontent.com/49945713/218988098-c354ef28-7d63-465c-ae18-ea6c17490cff.png)

--- a/src/modules/olMap/formats/kml.js
+++ b/src/modules/olMap/formats/kml.js
@@ -45,13 +45,10 @@ export const toKmlStyleProperties = (style) => {
 const sanitizeStyle = (styles) => {
 	const getFirstOrDefault = (styles) => {
 		const firstStyle = styles[0] ? styles[0] : styles && !Array.isArray(styles) ? styles : null;
-
 		return firstStyle ?? new Style();
 	};
 
 	const style = getFirstOrDefault(styles);
-	//const style = styles[0] ? styles[0].clone() : styles && !Array.isArray(styles) ? styles.clone() : new Style();
-
 	const kmlStyleProperties = toKmlStyleProperties(style);
 
 	if (kmlStyleProperties.image instanceof CircleStyle) {

--- a/src/modules/olMap/formats/kml.js
+++ b/src/modules/olMap/formats/kml.js
@@ -43,7 +43,14 @@ export const toKmlStyleProperties = (style) => {
 };
 
 const sanitizeStyle = (styles) => {
-	const style = styles[0] ? styles[0].clone() : styles && !Array.isArray(styles) ? styles.clone() : new Style();
+	const getFirstOrDefault = (styles) => {
+		const firstStyle = styles[0] ? styles[0] : styles && !Array.isArray(styles) ? styles : null;
+
+		return firstStyle ?? new Style();
+	};
+
+	const style = getFirstOrDefault(styles);
+	//const style = styles[0] ? styles[0].clone() : styles && !Array.isArray(styles) ? styles.clone() : new Style();
 
 	const kmlStyleProperties = toKmlStyleProperties(style);
 

--- a/src/modules/olMap/handler/measure/OlMeasurementHandler.js
+++ b/src/modules/olMap/handler/measure/OlMeasurementHandler.js
@@ -186,7 +186,7 @@ export class OlMeasurementHandler extends OlLayerHandler {
 				if (this._measureState.type === InteractionStateType.DRAW) {
 					setSelection([event.feature.getId()]);
 				}
-				this._save();
+				saveDebounced();
 			};
 			this._mapListeners.push(layer.getSource().on('addfeature', setSelectedAndSave));
 			this._mapListeners.push(layer.getSource().on('changefeature', () => saveDebounced()));

--- a/src/modules/toolbox/components/measureToolContent/MeasureToolContent.js
+++ b/src/modules/toolbox/components/measureToolContent/MeasureToolContent.js
@@ -173,7 +173,7 @@ export class MeasureToolContent extends AbstractToolContent {
 		const translate = (key) => this._translationService.translate(key);
 		const getTranslatedSpan = (key) => html`<span>${unsafeHTML(translate(key))}</span>`;
 		const getMeasurementModeMessage = (mode) => getTranslatedSpan('toolbox_measureTool_measure_' + mode);
-		return this._environmentService.isTouch() ? getMeasurementModeMessage(mode) : nothing;
+		return this._environmentService.isTouch() && mode ? getMeasurementModeMessage(mode) : nothing;
 	}
 
 	async _copyValueToClipboard(value, measure) {

--- a/test/modules/olMap/handler/measure/OlMeasurementHandler.test.js
+++ b/test/modules/olMap/handler/measure/OlMeasurementHandler.test.js
@@ -1009,8 +1009,8 @@ describe('OlMeasurementHandler', () => {
 	});
 
 	describe('when storing layer', () => {
+		const afterDebounceDelay = OlMeasurementHandler.Debounce_Delay + 100;
 		describe('debouncing takes place', () => {
-			const afterDebounceDelay = OlMeasurementHandler.Debounce_Delay + 100;
 			beforeEach(function () {
 				jasmine.clock().install();
 			});
@@ -1018,7 +1018,7 @@ describe('OlMeasurementHandler', () => {
 			afterEach(function () {
 				jasmine.clock().uninstall();
 			});
-			it('stores twice after a single change of a feature', async () => {
+			it('stores once after a single change of a feature', async () => {
 				setup();
 				const classUnderTest = new OlMeasurementHandler();
 				const map = setupMap();
@@ -1031,15 +1031,15 @@ describe('OlMeasurementHandler', () => {
 				const feature = new Feature({ geometry: geometry });
 
 				classUnderTest.activate(map);
-				classUnderTest._vectorLayer.getSource().addFeature(feature); // -> first call of _save, caused by vectorsource:addfeature-event
-				feature.getGeometry().dispatchEvent('change'); // -> first call of debounced _save, caused by vectorsource:changefeature-event
+				classUnderTest._vectorLayer.getSource().addFeature(feature); // -> first call of debounced _save, caused by vectorsource:addfeature-event
+				feature.getGeometry().dispatchEvent('change'); // -> second call of debounced _save, caused by vectorsource:changefeature-event
 				jasmine.clock().tick(afterDebounceDelay);
 
-				expect(privateSaveSpy).toHaveBeenCalledTimes(2);
-				expect(storeSpy).toHaveBeenCalledTimes(2);
+				expect(privateSaveSpy).toHaveBeenCalledTimes(1);
+				expect(storeSpy).toHaveBeenCalledTimes(1);
 			});
 
-			it('stores twice after a feature removed', async () => {
+			it('stores once after a feature removed', async () => {
 				setup();
 				const classUnderTest = new OlMeasurementHandler();
 				const map = setupMap();
@@ -1050,18 +1050,18 @@ describe('OlMeasurementHandler', () => {
 					[1, 0]
 				]);
 				const feature = new Feature({ geometry: geometry });
-				feature.set('debug', 'stores twice after a feature removed');
+				feature.set('debug', 'stores once after a feature removed');
 
 				classUnderTest.activate(map);
-				classUnderTest._vectorLayer.getSource().addFeature(feature); // -> first call of _save, caused by vectorsource:addfeature-event
-				classUnderTest._vectorLayer.getSource().removeFeature(feature); // -> first call of debounced _save, caused by vectorsource:removefeature-event
+				classUnderTest._vectorLayer.getSource().addFeature(feature); // -> first call of debounced _save, caused by vectorsource:addfeature-event
+				classUnderTest._vectorLayer.getSource().removeFeature(feature); // -> second call of debounced _save, caused by vectorsource:removefeature-event
 				jasmine.clock().tick(afterDebounceDelay);
 
-				expect(privateSaveSpy).toHaveBeenCalledTimes(2);
-				expect(storeSpy).toHaveBeenCalledTimes(2);
+				expect(privateSaveSpy).toHaveBeenCalledTimes(1);
+				expect(storeSpy).toHaveBeenCalledTimes(1);
 			});
 
-			it('stores only twice after multiple changes of a feature', async () => {
+			it('stores only once after multiple changes of a feature', async () => {
 				setup();
 				const classUnderTest = new OlMeasurementHandler();
 				const map = setupMap();
@@ -1074,15 +1074,15 @@ describe('OlMeasurementHandler', () => {
 				const feature = new Feature({ geometry: geometry });
 
 				classUnderTest.activate(map);
-				classUnderTest._vectorLayer.getSource().addFeature(feature); // -> first call of _save, caused by vectorsource:addfeature-event
+				classUnderTest._vectorLayer.getSource().addFeature(feature); // -> call of debounced _save, caused by vectorsource:addfeature-event
 				feature.dispatchEvent('change'); // -> second call of debounced _save, caused by vectorsource:changefeature-event
 				feature.dispatchEvent('change');
 				feature.dispatchEvent('change');
 				feature.dispatchEvent('change');
 				jasmine.clock().tick(afterDebounceDelay);
 
-				expect(privateSaveSpy).toHaveBeenCalledTimes(2);
-				expect(storeSpy).toHaveBeenCalledTimes(2);
+				expect(privateSaveSpy).toHaveBeenCalledTimes(1);
+				expect(storeSpy).toHaveBeenCalledTimes(1);
 			});
 		});
 
@@ -1101,7 +1101,7 @@ describe('OlMeasurementHandler', () => {
 			});
 			classUnderTest._vectorLayer.getSource().addFeature(feature);
 
-			await TestUtils.timeout();
+			await TestUtils.timeout(afterDebounceDelay);
 			expect(storageSpy).toHaveBeenCalledWith(jasmine.any(String), FileStorageServiceDataTypes.KML);
 		});
 	});

--- a/test/modules/toolbox/components/measureToolContent/MeasureToolContent.test.js
+++ b/test/modules/toolbox/components/measureToolContent/MeasureToolContent.test.js
@@ -439,6 +439,17 @@ describe('MeasureToolContent', () => {
 				expect(subTextElement).toBeTruthy();
 				expect(subTextElement.textContent).toBe('toolbox_measureTool_measure_select');
 			});
+
+			it('shows no measurement sub-text for mode:[null]', async () => {
+				const state = {
+					measurement: { ...defaultMeasurementState, mode: null }
+				};
+				const element = await setup(state, touchConfig);
+				const subTextElement = element.shadowRoot.querySelector('.sub-text');
+
+				expect(subTextElement).toBeTruthy();
+				expect(subTextElement.textContent).toBe('');
+			});
 		});
 	});
 });

